soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4059,22 +4181,90 @@ void DrawRandoEditor(bool& open) {
+                    PaddedSeparator();
+
+                    // Gerudo Keys
+                    ImGui::Text(Settings::GerudoKeys.GetName().c_str());
+                    InsertHelpHoverText(
+                        "Vanilla - Thieve's Hideout Keys will appear in their vanilla locations.\n"
+                        "\n"
+                        "Any dungeon - Thieve's Hideout Keys can only appear inside of any dungon.\n"
+                        "\n"
+                        "Overworld - Thieve's Hideout Keys can only appear outside of dungeons.\n"
+                        "\n"
+                        "Anywhere - Thieve's Hideout Keys can appear anywhere in the world."
+                    );
+                    SohImGui::EnhancementCombobox("gRandomizeGerudoKeys", randoShuffleGerudoFortressKeys, 4, 0);
+                    PaddedSeparator();
+
+                    // Boss Keysanity
+                    ImGui::Text(Settings::BossKeysanity.GetName().c_str());
+                    InsertHelpHoverText(
+                        "Start with - You will start with Boss keys from all dungeons.\n"
+                        "\n"
+                        "Vanilla - Boss Keys will appear in their vanilla locations.\n"
+                        "\n"
+                        "Own dungeon - Boss Keys can only appear in their respective dungeon.\n"
+                        "\n"
+                        "Any dungeon - Boss Keys can only appear inside of any dungon.\n"
+                        "\n"
+                        "Overworld - Boss Keys can only appear outside of dungeons.\n"
+                        "\n"
+                        "Anywhere - Boss Keys can appear anywhere in the world."
+                    );  
+                    SohImGui::EnhancementCombobox("gRandomizeBossKeysanity", randoShuffleBossKeys, 6, 2);
+                    PaddedSeparator();
+
                     // Ganon's Boss Key
                     ImGui::Text(Settings::GanonsBossKey.GetName().c_str());
                     InsertHelpHoverText(
-                        "Vanilla - Key will appear in the vanilla location.\n"
+                        "Vanilla - Ganon's Boss Key will appear in the vanilla location.\n"
                         "\n"
-                        "Own dungeon - Key can appear anywhere inside Ganon's Castle.\n"
+                        "Own dungeon - Ganon's Boss Key can appear anywhere inside Ganon's Castle.\n"
                         "\n"
                         "Start with - Places Ganon's Boss Key in your starting inventory."
+                        "\n"
+                        "Any dungeon - Ganon's Boss Key Key can only appear inside of any dungon.\n"
+                        "\n"
+                        "Overworld - Ganon's Boss Key Key can only appear outside of dungeons.\n"
+                        "\n"
+                        "Anywhere - Ganon's Boss Key Key can appear anywhere in the world."
```

Comment:
There's a few more LACS options here, but I have no idea what they do, probably need to copy descriptions from 3drando

Reply:
![image](https://user-images.githubusercontent.com/50862920/183653046-d50daa4f-23a0-4d29-ab56-7ea443ad3e09.png)

From 3drando

Reply:
i feel like those options are for doing rainbow bridge open but then you get the boss key from the LACS which is changed to use the different requirements (instead of just spirit/shadow medallions)

i don't think those need to be implemented as part of keysanity (and it'd probably be better to have that as it's own PR for clarity's sake)

Reply:
@Flax4 i hadn't refreshed quickly enough to see that when writing my comment, thanks for confirming